Scenes/Tavern/Publician.gd

```diff
@@ -0,0 +1,199 @@
+		selected_quest = item_list.get_selected_items()[0]
+	
+func load_quests():
+	var save_file = File.new()
+	if not save_file.file_exists(SAVE_PATH):
+		print("Nie znaleziono pliku!!!")
+		create_default_quests()
+		return
+	
+	save_file.open(SAVE_PATH, File.READ)
+	var data = JSON.parse(save_file.get_as_text()).result;
+
+	for index in data.keys():
+		if data[index]["type"] == QuestType.kill_dwarver:
+			quests.append(QuestKill.new(data[index]["count"]))
+		if data[index]["type"] == QuestType.spend_xp:
+			quests.append(QuestSpendXp.new(data[index]["count"]))
+		if data[index]["type"] == QuestType.spend_gold:
+			quests.append(QuestSpendXp.new(data[index]["count"]))
+	
+	load_list_panel()
+
+func save_quests():
+	var save_dict = { }
+	for quest in quests:
+		save_dict[quests.find(quest)] = quest.data
+	
+	var save_file = File.new()
+	save_file.open(SAVE_PATH, File.WRITE)
+	save_file.store_line(to_json(save_dict))
+	save_file.close()
+
+func create_default_quests():
+	add_kill_dwarves_quest()
+	add_spend_gold_quest()
+	add_kill_dwarves_quest()
+	add_spend_xp_quest()
+	add_spend_xp_quest()
+	add_spend_gold_quest()
+	add_kill_dwarves_quest()
+	add_spend_gold_quest()
+	add_kill_dwarves_quest()
+	add_spend_xp_quest()
+	add_spend_xp_quest()
+	add_spend_gold_quest()
+
+func load_list_panel():
+	for index in quests.size():
+		item_list.add_item(String(quests[index].data["title"]),dwarver_cion)
+
+class QuestKill:
+	var data = {
+		title = "Zabij x krasnali",
+		gold_reward = 100.0,
+		xp_reward = 40.0,
+		count = 4,
+		amount = 0,
+		type = QuestType.kill_dwarver
+		}
+	
+	func _init(Count):
+		data["count"] = Count
+		data["title"] = "Zabij krasnale: "+String(Count)	
+		data["gold_reward"] =  stepify(float(Count) * float(Count) * randf()*2,2)
+		data["xp_reward"] = stepify(Count * (randf()*6+1),2)
+
+class QuestSpendXp:
+	var data = {
+		title = "Wydaj x punktow doswiadczenia",
+		gold_reward = 100.0,
+		xp_reward = 40.0,
+		count = 4,
+		amount = 0,
+		type = QuestType.spend_xp
+		}
+	
+	func _init(Count):
+		data["count"] = Count
+		data["title"] = "Wydaj "+String(Count)+" punktow doswiadczenia"
+		data["gold_reward"] =  Count
+		data["xp_reward"] = Count*5
+
+class QuestSpendGold:
+	var data = {
+		title = "Wydaj x golda",
+		gold_reward = 100.0,
+		xp_reward = 40.0,
+		count = 4,
+		amount = 0,
+		type = QuestType.spend_gold
+		}
+	
+	func _init(Count):
+		data["count"] = Count
+		data["title"] = "Wydaj "+String(Count)+" golda"
+		data["gold_reward"] =  Count
+		data["xp_reward"] = Count*5
+
+func on_spend_xp(spent_xp):
+	for index in quests.size():
+		if index == selected_quest:
+			if quests[index].data["type"] == QuestType.spend_xp:
+				quests[index].data["amount"] += spent_xp
+				if quests[index].data["amount"] >= quests[index].data["count"]:
+					give_reward(quests[index].data["gold_reward"], quests[index].data["xp_reward"], index)
+					delete_quest(index)
+					add_spend_xp_quest()
+
+func on_kill_dwarver():
```

Comment:
In those three functions:
`on_spend_xp`, `on_kill_dwarver`, `on_spend_gold`, there are so many repetable code. Should you try to do this in one function, with (for example) match statement somwhere and additional parameter in this function?